astropy/units/physical.py

```diff
@@ -1,133 +1,425 @@
+        return item in self.as_set
+
+    def __repr__(self):
+        return repr(self.as_string) if len(self.as_set) == 1 else self.as_string
+
+    def __str__(self):
+        return self.as_string
+
+    @staticmethod
+    def _identify_unit_from_unit_or_physical_type(obj) -> core.Unit:
+        """
+        If a unit is passed in, return that unit.  If a physical type is
+        passed in, return a unit that corresponds to that physical type.
+        """
+        if isinstance(obj, core.UnitBase):
+            return obj
+        elif isinstance(obj, PhysicalType):
+            return obj._unit
+        else:
+            raise TypeError("Expecting a unit or a physical type")
+
+    def __mul__(self, other):
+        other_unit = self._identify_unit_from_unit_or_physical_type(other)
+        other_unit = _replace_temperatures_with_kelvin(other_unit)
+        new_unit = self._unit * other_unit
+        return new_unit.physical_type
+
+    def __rmul__(self, other):
+        return self.__mul__(other)
+
+    def __truediv__(self, other):
+        other_unit = self._identify_unit_from_unit_or_physical_type(other)
+        other_unit = _replace_temperatures_with_kelvin(other_unit)
+        new_unit = self._unit / other_unit
+        return new_unit.physical_type
+
+    def __rtruediv__(self, other):
+        other_unit = self._identify_unit_from_unit_or_physical_type(other)
+        other_unit = _replace_temperatures_with_kelvin(other_unit)
+        new_unit = other_unit / self._unit
+        return new_unit.physical_type
+
+    def __pow__(self, power):
+        if not isinstance(power, numbers.Real):
+            raise TypeError(f"{power} is not a real number")
+        return (self._unit ** power).physical_type
+
+
+def def_physical_type(unit: core.UnitBase, name: Union[str, Set[str]]):
+    """
+    Add a mapping between a unit and the corresponding physical type(s).
+
+    Parameters
+    ----------
+    unit : u.Unit
+        The unit to be represented by the physical type.
+
+    name : `str` or `set` of `str`
+        A `str` representing the name of the physical type of the unit,
+        or a `set` containing strings that represent one or more names
+        of physical types.
+    """
+
+    if name in ("unknown", {"unknown"}) or "unknown" in name:
+        raise ValueError("Unable to uniquely define an unknown physical type")
+
+    physical_type_id = unit._get_physical_type_id()
+    if physical_type_id in _physical_unit_mapping:
         raise ValueError(
-            f"{r!r} ({name!r}) already defined as {_physical_unit_mapping[r]!r}")
-    _physical_unit_mapping[r] = name
-    _unit_physical_mapping[name] = r
+            f"{physical_type_id!r} ({name!r}) already defined as "
+            f"{_physical_unit_mapping[physical_type_id]!r}"
+        )
+
+    physical_type = PhysicalType(unit, name)
 
+    _physical_unit_mapping[physical_type_id] = physical_type
 
-def get_physical_type(unit):
+    for ptype in physical_type.as_set:
```

Comment:
With `__iter__` defined as above, you can just iterate directly `in physical_type`.